third_party/move/move-vm/runtime/src/storage/module_storage.rs

```diff
@@ -101,6 +113,90 @@ pub trait ModuleStorage: WithRuntimeEnvironment {
         address: &AccountAddress,
         module_name: &IdentStr,
     ) -> VMResult<Option<Arc<Module>>>;
+
+    /// Returns the verified module. If it does not exist, a linker error is returned. All other
+    /// errors are mapped using [expect_no_verification_errors] - since on-chain code should not
+    /// fail bytecode verification.
+    fn fetch_existing_verified_module(
+        &self,
+        address: &AccountAddress,
+        module_name: &IdentStr,
+    ) -> VMResult<Arc<Module>> {
+        self.fetch_verified_module(address, module_name)
+            .map_err(expect_no_verification_errors)?
+            .ok_or_else(|| module_linker_error!(address, module_name))
+    }
+
+    /// Returns a struct type corresponding to the specified name. The module containing the struct
+    /// will be fetched and cached beforehand.
+    fn fetch_struct_ty(
+        &self,
+        address: &AccountAddress,
+        module_name: &IdentStr,
+        struct_name: &IdentStr,
+    ) -> PartialVMResult<Arc<StructType>> {
+        let module = self
+            .fetch_existing_verified_module(address, module_name)
+            .map_err(|err| err.to_partial())?;
+        Ok(module
+            .struct_map
+            .get(struct_name)
+            .and_then(|idx| module.structs.get(*idx))
+            .ok_or_else(|| {
+                PartialVMError::new(StatusCode::TYPE_RESOLUTION_FAILURE).with_message(format!(
+                    "Struct {}::{}::{} does not exist",
+                    address, module_name, struct_name
+                ))
+            })?
+            .definition_struct_type
+            .clone())
+    }
+
+    /// Returns a runtime type corresponding to the specified type tag (file format type
+    /// representation). If a struct type is constructed, the module containing the struct
+    /// definition is fetched and cached.
+    fn fetch_ty(&self, ty_tag: &TypeTag) -> PartialVMResult<Type> {
+        // TODO(loader_v2): Loader V1 uses VMResults everywhere, but partial VM errors
+        //                  seem better fit. Here we map error to VMError to reuse existing
+        //                  type builder implementation, and then strip the location info.
+        self.runtime_environment()
+            .vm_config()
+            .ty_builder
+            .create_ty(ty_tag, |st| {
+                self.fetch_struct_ty(
+                    &st.address,
+                    st.module.as_ident_str(),
+                    st.name.as_ident_str(),
+                )
+                .map_err(|err| err.finish(Location::Undefined))
+            })
+            .map_err(|err| err.to_partial())
+    }
+
+    /// Returns the function definition corresponding to the specified name, as well as the module
+    /// where this function is defined. The returned module is also verified.
+    fn fetch_function_without_ty_args(
```

Comment:
nit: why explicitly `without_ty_args`? in the name

Reply:
I think this should be called `fetch_function_definition` then? Basically types might be generic, this is not instantiated function yet, and to get it you do not provide type args.